src/mame/sony_news/cxd8442q.cpp

```diff
@@ -0,0 +1,364 @@
+// license:BSD-3-Clause
+// copyright-holders:Brice Onken
+
+/*
+ * Sony CXD8442Q WSC-FIFOQ APbus FIFO and DMA controller
+ *
+ * The FIFOQ is an APbus DMA controller designed for interfacing some of the simpler and lower speed peripherals
+ * to the APbus while providing DMA capabilities. Each FIFO chip can support up to 4 devices. There is no
+ * documentation avaliable for these chips (that I have been able to find, anyways), so this implements the bare minimum
+ * needed to satisfy the monitor ROM (for booting off of floppy drives) and NEWS-OS (for async serial communication).
+ * I'm sure there is a lot of missing or hardware inaccurate functionality here - this was all derived from running stuff
+ * and observing the debugger and emulator log files. Additionally, the way this is coded makes it interrupt pretty much whenever
+ * data is avaliable. The real hardware probably buffers this more.
+ *
+ * The NWS-5000X uses two of these chips to drive the following peripherals:
+ *  - Floppy disk controller
+ *  - Sound
+ *  - Asynchronous serial communication
+ *  and potentially more.
+ *
+ * TODO:
+ *  - Hardware-accurate behavior of the FIFO - this is a best guess.
+ *  - Actual clock rate
+ *  - Additional features and registers
+ */
+
+#include "emu.h"
+#include "cxd8442q.h"
+#include "logmacro.h"
+
+DEFINE_DEVICE_TYPE(CXD8442Q, cxd8442q_device, "cxd8442q", "Sony CXD8442Q WSC-FIFOQ")
+
+namespace
+{
+	// 128KiB used as the FIFO RAM (can be divided up to 4 regions, 1 per channel)
+	static constexpr int FIFO_MAX_RAM_SIZE = 0x20000;
```

Comment:
There’s no point using `static` linkage inside an anonymous namespace – stuff won’t be externally accessible anyway.